tests/Giraffe.Tests/HttpHandlerTests.fs

```diff
@@ -1187,4 +1187,11 @@ let ``Get "/flex" without an Accept header returns a JSON object`` () =
     | Some ctx ->
         let body = getBody ctx
         Assert.Equal(expected, body)
-        Assert.Equal("application/json", ctx.HttpContext.Response |> getContentType)
\ No newline at end of file
+        Assert.Equal("application/json", ctx.HttpContext.Response |> getContentType)
+
+[<Fact>]
+let ``Warbler function should execute inner function`` () =
+    let inner (a: int) = a.ToString()
+    let warbled = warbler (fun (a:int) -> inner)
+    let result = warbled 42
+    Assert.Equal("42", result)
```

Comment:
Something like 
```fsharp
[<Fact>]
let ``Warbler function should execute inner function twice`` () =
    let inner = Guid.NewGuid().ToString()
    let warbled = warbler (fun _ -> inner)

    let r1 = inner
    let r2 = inner
    let r3 = warbled
    let r4 = warbled

    Assert.Equal(r1, r2)
    Assert.NotEqual(r3, r4)
```

Doesn't compile since inner is a `string` and it should be `a' -> b'`. Adding unit to inner will execute twice so I'm not sure how to proceed.